feat(replays): Display status code in Replay Network Table

jesus4497 opened the pull request:
<!-- Describe your PR here. -->
Added status code value to the network table. Closes #37926 

![image](https://user-images.githubusercontent.com/39612839/185493934-4b81bedd-d744-4294-986a-710d885bd969.png)

![image](https://user-images.githubusercontent.com/39612839/185493873-9dfcf9c6-6fbd-48ba-8ad7-6d4e8f680507.png)

![image](https://user-images.githubusercontent.com/39612839/185472196-85e43084-8fd3-4a34-8a6b-2a97fa804435.png)


<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

jesus4497 committed:
feat: add status code value

jesus4497 committed:
style: shrink empty status text

jesus4497 committed:
style: status column alignment

danecando commented:
LGTM only comment I have is the text 'Missing status' since the column is status could just be 'missing' or 'n/a'. I'd wait to see if anyone else agrees though

billyvg commented on static/app/views/replays/detail/network/index.tsx:
```diff
@@ -248,10 +255,17 @@ const StyledPanelTable = styled(PanelTable)<{columns: number}>`
 `;
 
-const StatusPlaceHolder = styled(Placeholder)`
+const StatusContainer = styled('div')`
```
I would name this `StatusBubble`

ryan953 replied:
Should we drop the bubble? 

Latest designs have dropped it:
<img width="656" alt="Screen Shot 2022-08-18 at 9 13 48 AM" src="https://user-images.githubusercontent.com/187460/185443820-d9f925d1-e623-4736-850e-d157ab4bc72a.png">

jesus4497 committed:
style: add new styles to the table

jesus4497 commented:
![image](https://user-images.githubusercontent.com/39612839/185472196-85e43084-8fd3-4a34-8a6b-2a97fa804435.png)

Updated! Also changed the styles to resemble those of the new figma styles, as this was a quick thing to do I thought on doing it in this same PR.

Should I also change the status code to the "OK" word as it is shown in the figma?

billyvg commented on static/app/views/replays/detail/network/index.tsx:
```diff
@@ -237,23 +252,25 @@ const StyledPanelTable = styled(PanelTable)<{columns: number}>`
     line-height: 16px;
+    text-transform: none;
+
+    ${/* sc-selector */ UnstyledButton} {
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+    }
```
Instead of nesting this, we should create a new styled component:

```suggestion
const UnstyledHeaderButton = styled(UnstyledButton)`...`
```

I would do the same with `PanelTableHeader` too.

jesus4497 replied:
Done! for `PanelTableHeader ` I think I cannot do the same as this component is being used inside the `PanelTable` component

billyvg replied:
Ah right

jesus4497 committed:
style: add button styled component

jesus4497 committed:
Merge branch 'master' into feat/replays/37926-jesus

billyvg reviewed (approved):
Going to merge this when CI passes, but it needs a follow-up maybe to fix sorting by `status` -- when you sort, the rows with status are always at the top.

ryan953 reviewed (approved):
awesome, needs a fresh screen shot and we'll get it merged!

ryan953 commented:
> Should I also change the status code to the "OK" word as it is shown in the figma?

@jesus4497 We don't need to change these values to match the figma mocks.  We want to show whatever value is inside of `network.data.statusCode`. 
Right now `statusCode` contains values that are `string | number`. In the future if is `string | number | boolean` then we'll do special rendering for those booleans.

In Figma we were just getting a sense of it could look like.
